interfaces/core.go

```diff
@@ -113,6 +113,15 @@ type Info struct {
 	Slots   []*snap.SlotInfo
 }
 
+// ConnectionInfo describes a connection between a plug and a slot for the user.
+// Either the plug or the slot may be nil to represent a slot/plug that's not connected.
```

Comment:
The interface here feels a bit awkward. It means every single place we use this type we'll need to check whether its fields are set, and which of them are set. It's in fact not a *connection* info. I think we should separate those two ideas, and have connections represented really as connections, and plugs and slots really as plugs and slots. It also means the latter information won't have duplicates.